compiler/rustc_builtin_macros/src/derive.rs

```diff
@@ -33,14 +33,15 @@ impl MultiItemModifier for Expander {
             ecx.resolver.resolve_derives(ecx.current_expansion.id, ecx.force_mode, &|| {
                 let template =
                     AttributeTemplate { list: Some("Trait1, Trait2, ..."), ..Default::default() };
-                let attr =
-                    attr::mk_attr_outer(&sess.parse_sess.attr_id_generator, meta_item.clone());
-                validate_attr::check_builtin_attribute(
+                validate_attr::check_builtin_meta_item(
                     &sess.parse_sess,
-                    &attr,
+                    &meta_item,
+                    ast::AttrStyle::Outer,
                     sym::derive,
                     template,
                 );
+                let attr =
+                    attr::mk_attr_outer(&sess.parse_sess.attr_id_generator, meta_item.clone());
```

Comment:
This `attr` is only used for calling `.meta_item_list()` below.
The existing `meta_item` should be enough to do the same thing without creating `attr`.

Reply:
I tried that, and it turned out to be painful enough that I gave up, I think because it required working at the token level. I could try again, I guess.

Reply:
You were right, this works, though there was a quirk with an error case in a `ui` test. Details in the commit message in the new commit.